src/solvers/bitwuzla/bitwuzla_conv.cpp

```diff
@@ -0,0 +1,920 @@
+    break;
+
+  case SMT_SORT_ARRAY:
+    // Fedor: not entirely sure about this one here
+    //node = bitwuzla_mk_const_array(bitw, to_solver_smt_sort<BitwuzlaSort*>(s)->s, name.c_str());
+    node = bitwuzla_mk_const(
+      bitw, to_solver_smt_sort<BitwuzlaSort *>(s)->s, name.c_str());
+    break;
+
+  default:
+    std::cerr << "Unknown type for symbol\n";
+    abort();
+  }
+
+  smt_astt ast = new_ast(node, s);
+
+  symtable.insert(symtable_type::value_type(name, ast));
+
+  return ast;
+}
+
+smt_astt
+bitwuzla_convt::mk_extract(smt_astt a, unsigned int high, unsigned int low)
+{
+  smt_sortt s = mk_bv_sort(high - low + 1);
+  const bitw_smt_ast *ast = to_solver_smt_ast<bitw_smt_ast>(a);
+  //BoolectorNode *b = boolector_slice(btor, ast->a, high, low);
+  BitwuzlaTerm *b = bitwuzla_mk_term1_indexed2(
+    bitw, BITWUZLA_KIND_BV_EXTRACT, ast->a, high, low);
+  return new_ast(b, s);
+}
+
+smt_astt bitwuzla_convt::mk_sign_ext(smt_astt a, unsigned int topwidth)
+{
+  smt_sortt s = mk_bv_sort(a->sort->get_data_width() + topwidth);
+  const bitw_smt_ast *ast = to_solver_smt_ast<bitw_smt_ast>(a);
+  //BoolectorNode *b = boolector_sext(btor, ast->a, topwidth);
+  BitwuzlaTerm *b = bitwuzla_mk_term1_indexed1(
+    bitw, BITWUZLA_KIND_BV_SIGN_EXTEND, ast->a, topwidth);
+  return new_ast(b, s);
+}
+
+smt_astt bitwuzla_convt::mk_zero_ext(smt_astt a, unsigned int topwidth)
+{
+  smt_sortt s = mk_bv_sort(a->sort->get_data_width() + topwidth);
+  const bitw_smt_ast *ast = to_solver_smt_ast<bitw_smt_ast>(a);
+  //BoolectorNode *b = boolector_uext(btor, ast->a, topwidth);
+  BitwuzlaTerm *b = bitwuzla_mk_term1_indexed1(
+    bitw, BITWUZLA_KIND_BV_ZERO_EXTEND, ast->a, topwidth);
+  return new_ast(b, s);
+}
+
+smt_astt bitwuzla_convt::mk_concat(smt_astt a, smt_astt b)
+{
+  smt_sortt s =
+    mk_bv_sort(a->sort->get_data_width() + b->sort->get_data_width());
+
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_BV_CONCAT,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    s);
+}
+
+smt_astt bitwuzla_convt::mk_ite(smt_astt cond, smt_astt t, smt_astt f)
+{
+  assert(cond->sort->id == SMT_SORT_BOOL);
+  assert(t->sort->get_data_width() == f->sort->get_data_width());
+
+  return new_ast(
+    bitwuzla_mk_term3(
+      bitw,
+      BITWUZLA_KIND_ITE,
+      to_solver_smt_ast<bitw_smt_ast>(cond)->a,
+      to_solver_smt_ast<bitw_smt_ast>(t)->a,
+      to_solver_smt_ast<bitw_smt_ast>(f)->a),
+    t->sort);
+}
+
+bool bitwuzla_convt::get_bool(smt_astt a)
+{
+  const bitw_smt_ast *ast = to_solver_smt_ast<bitw_smt_ast>(a);
+  BitwuzlaTerm *true_term = bitwuzla_mk_true(bitw);
+  BitwuzlaTerm *false_term = bitwuzla_mk_false(bitw);
+  BitwuzlaTerm *bitw_term = bitwuzla_get_value(bitw, ast->a);
+
+  if(bitw_term == true_term)
+    return true;
+
+  if(bitw_term == false_term)
+    return false;
+
+  std::cerr << "Can't get boolean value from Bitwuzla\n";
+  abort();
+  /*
```

Comment:
This looks wrong, you are comparing pointers.

Reply:
This is an experimental implementation of the method for now while Bitwuzla developers are adding methods for obtaining string representations of Bitwuzla nodes.